handlers/whatsapp/whatsapp.go

```diff
@@ -134,6 +134,13 @@ func (h *handler) receiveEvent(ctx context.Context, channel courier.Channel, w h
 	// first deal with any received messages
 	for _, msg := range payload.Messages {
 
+		if msg.ID != "" {
```

Comment:
The best place to check external ID is seen is https://github.com/nyaruka/courier/pull/227/files#diff-3b05f66223be3994d25cf04fda0cfca0L175 and use `alreadyWritten` field to control the if we need to write to the DB

Reply:
Thanks for the review. 

Do you mean inside the `NewIncomingMsg` function? I kept it outside to only do it in the WA handler.

Reply:
Not inside rather after the call on that line, I agree we need this in the WA handle only

Reply:
Ya, main thing here I think is that we do want to echo the message back from a courier contract perspective just with the UUID of the previous message.